### PR TITLE
LPS-192657 Use new Import view for page templates

### PR DESCRIPTION
**Pull request by sandrodw3**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by sandrodw3** [comment]
ci:test:sf

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		long layoutPageTemplateCollectionId = ParamUtil.getLong(
+			resourceRequest, "layoutPageTemplateCollectionId");
+
+		UploadPortletRequest uploadPortletRequest =
+			_portal.getUploadPortletRequest(resourceRequest);
+
+		File file = uploadPortletRequest.getFile("file");
+
+		boolean overwrite = ParamUtil.getBoolean(
+			resourceRequest, "overwrite", true);
```
Comment (lfbesada) [inline modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java:69]: I think you could inline the file and the override. What do you think?

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+		JSONPortletResponseUtil.writeJSON(
+			resourceRequest, resourceResponse, jsonObject);
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		ImportMVCResourceCommand.class);
+
+	@Reference
+	private FragmentsImporter _fragmentsImporter;
```
Comment (lfbesada) [inline modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java:124]: I think that it is unused...

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/display/context/ImportDisplayContext.java [hunk 3]
```diff
@@ -253,6 +260,30 @@ public String getDialogType() {
...
+			() -> {
+				ResourceURL importURL = _renderResponse.createResourceURL();
+
+				importURL.setParameter(
+					"layoutPageTemplateCollectionId",
+					ParamUtil.getString(
+						_httpServletRequest, "layoutPageTemplateCollectionId"));
+				importURL.setResourceID("/layout_page_template_admin/import");
+
+				return importURL.toString();
```
Comment (lfbesada) [inline modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/display/context/ImportDisplayContext.java:282]: What do you think about to use here ResourceURLBuilder?

**Review by lfbesada — CHANGES_REQUESTED** [review]
Hi @victorg1991 
Thanks! It works great!
I have added a few comments about source formatting improvements
Regards

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java (hunk 2 above)
Comment (victorg1991) [reply modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java:124]: true :sweat:

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/display/context/ImportDisplayContext.java (hunk 3 above)
Comment (victorg1991) [reply modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/display/context/ImportDisplayContext.java:282]: It is like this in other parts of the module, but I'll update this one :)

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java (hunk 1 above)
Comment (victorg1991) [reply modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/ImportMVCResourceCommand.java:69]: I will :D

**Review by claraizquierdo — COMMENTED** [review]
Hi @sandrodw3 ,

great PR, very easy to review 🎉  Everything seems to be working fine but I couldn't test the results with warnings. Just one small detail that I know that wasn't introduced in this pull: The button text should be "Select File"/"Replace File" instead of "Select Files"/"Replace Files". Do you think we could fix it in this PR?

**Comment by sandrodw3** [comment]
> Hi @sandrodw3 ,
> 
> great PR, very easy to review tada Everything seems to be working fine but I couldn't test the results with warnings. Just one small detail that I know that wasn't introduced in this pull: The button text should be "Select File"/"Replace File" instead of "Select Files"/"Replace Files". Do you think we could fix it in this PR?

Sure!! Thanks a lot @claraizquierdo 🥰

**Comment by claraizquierdo** [comment]
Hi again @sandrodw3 , another thing that I don't know if is intended is that the link takes me to this url: https://liferay.dev/

**Comment by sandrodw3** [comment]
> Hi again @sandrodw3 , another thing that I don't know if is intended is that the link takes me to this url: https://liferay.dev/

That's the link we already had, it looked also weird to me, but 🤷‍♂️

**Comment by sandrodw3** [comment]
ci:forward

**Review by lfbesada — APPROVED** [review]

**Review by claraizquierdo — APPROVED** [review]

**Comment by sandrodw3** [comment]
ci:forward

**Comment by liulukiwi** [comment]
Pushed to BChan.
https://github.com/brianchandotcom/liferay-portal/pull/138993

**Pull request closed without merging**